File: src/main/java/zanoccio/javakit/ArrayUtilities.java

package zanoccio.javakit;

public class ArrayUtilities {
	public static String toString(byte[] array) {
		StringBuilder sb = new StringBuilder();

		for (int i = 0; i < array.length; i++) {
			if (i > 0)
				sb.append('-');

			if (array[i] < 0)
				sb.append(127 + -array[i]);
			else
				sb.append(array[i]);
		}

		return sb.toString();
	}
}

File: src/main/java/zanoccio/javakit/ArrayUtilities.scala

package zanoccio.javakit

object ArrayUtilities {
  def toString(array : Array[Byte]) = {
    val sb = new StringBuilder()

    for (i <- 0 until array.length) {
      if (i > 0)
        sb.append('-')

      if (array(i) < 0)
        sb.append(127 + -array(i))
      else
        sb.append(array(i))
    }

    sb.toString()
  }
}
